woocommerce/woocommerce-android
# [Issue 4614] Improve shipping labels validation scrolling

Closes: #4614


### Description

This pull request introduces a new feature to the address form in the shipping label creation workflow. The primary goal is to enhance the user experience by automatically scrolling to the first field that fails validation when the user attempts to submit the form with invalid data.

**Key Changes:**

1. Scrolling Logic: Added logic to `EditShippingLabelAddressViewModel` to determine the first field with a validation error. When the user clicks the "Done" button, and if any field validation fails, the form will now scroll to the first invalid field, making it easier for the user to identify and correct the error.

2. Event Triggering: Modified the `onDoneButtonClicked` method to trigger a `ScrollToFirstErrorField` event if there are any validation errors. This event includes the field that needs attention and a flag indicating whether the state field should be displayed as a spinner.

4. ViewModel Testing: Expanded the `EditShippingLabelAddressViewModelTest` suite to include tests for the new scrolling functionality. These tests verify that the correct events are triggered based on the form's validation state and that the form behaves as expected under various scenarios.

### Testing instructions
1. Create an order eligible for shipping label creation.
2. Open the order in the app.
3. Click on "Create shipping label"
4. Ensure that you remove the data from a few required fields like name, company, address. 
5. Click done and notice the app scrolls to the first error field. 

### Images/gif


<kbd><video src="https://github.com/woocommerce/woocommerce-android/assets/1509205/ed19c45e-d790-4f12-98cc-7f7d97a87825" width="320"></kbd>
